lib/rubygems/remote_fetcher.rb

```diff
@@ -414,4 +412,17 @@ def pools_for proxy
       @pools[proxy] ||= Gem::Request::ConnectionPools.new proxy, @cert_files
     end
   end
+
+  def s3_source_auth(uri)
```

Comment:
Could you add some newlines to break this up into logical chunks? Just adding a single blank line after the early `return` and each `raise` call in this function is sufficient imo. :slightly_smiling_face: